Core/include/Acts/Vertexing/SeedVertexFinder.ipp

```diff
@@ -0,0 +1,656 @@
+    ACTS_ERROR("value of maxXYdeviation was not initialized");
+  }
+  if (std::isnan(cfg.maxXYZdeviation)) {
+    ACTS_ERROR("value of maxXYZdeviation was not initialized");
+  }
+  if (std::isnan(cfg.minTheta)) {
+    ACTS_ERROR("value of minTheta was not initialized");
+  }
+  if (std::isnan(cfg.rMinNear)) {
+    ACTS_ERROR("value of rMinNear was not initialized");
+  }
+  if (std::isnan(cfg.rMaxNear)) {
+    ACTS_ERROR("value of rMaxNear was not initialized");
+  }
+  if (std::isnan(cfg.rMinMiddle)) {
+    ACTS_ERROR("value of rMinMiddle was not initialized");
+  }
+  if (std::isnan(cfg.rMaxMiddle)) {
+    ACTS_ERROR("value of rMaxMiddle was not initialized");
+  }
+  if (std::isnan(cfg.rMinFar)) {
+    ACTS_ERROR("value of rMinFar was not initialized");
+  }
+  if (std::isnan(cfg.rMaxFar)) {
+    ACTS_ERROR("value of rMaxFar was not initialized");
+  }
+  if (std::isnan(cfg.numPhiSlices)) {
+    ACTS_ERROR("value of numPhiSlices was not initialized");
+  }
+  if (cfg.numPhiSlices < 3) {
+    ACTS_INFO("value of numPhiSlices is "
+              << cfg.numPhiSlices
+              << ", which is less than 3. There will be duplicate triplets.");
+  }
+  if (std::isnan(cfg.useFracPhiSlices)) {
+    ACTS_ERROR("value of useFracPhiSlices was not initialized");
+  }
+  if (cfg.useFracPhiSlices < 0. || cfg.useFracPhiSlices >= 1.) {
+    ACTS_ERROR("value of useFracPhiSlices is "
+               << cfg.useFracPhiSlices
+               << ", allowed values are between 0 and 1");
+  }
+  if (std::isnan(cfg.numZSlices)) {
+    ACTS_ERROR("value of numZSlices was not initialized");
+  }
+  if (std::isnan(cfg.useFracZSlices)) {
+    ACTS_ERROR("value of useFracZSlices was not initialized");
+  }
+  if (cfg.useFracZSlices < 0. || cfg.useFracZSlices >= 1.) {
+    ACTS_ERROR("value of useFracZSlices is "
+               << cfg.useFracZSlices << ", allowed values are between 0 and 1");
+  }
+  if (std::isnan(cfg.maxAbsZ)) {
+    ACTS_ERROR("value of maxAbsZ was not initialized");
+  }
+  if (std::isnan(cfg.maxZPosition)) {
+    ACTS_ERROR("value of maxZPosition was not initialized");
+  }
+  if (std::isnan(cfg.maxRPosition)) {
+    ACTS_ERROR("value of maxRPosition was not initialized");
+  }
+  if (cfg.minimalizeWRT != "planes" && cfg.minimalizeWRT != "rays") {
+    ACTS_ERROR("value of minimalizeWRT is "
+               << cfg.minimalizeWRT
+               << ", allowed values are \"planes\" or \"rays\" ");
+  }
+  if (std::isnan(cfg.maxIterations)) {
+    ACTS_ERROR("value of maxIterations was not initialized");
+  }
+  if (std::isnan(cfg.removeFraction)) {
+    ACTS_ERROR("value of removeFraction was not initialized");
+  }
+  if (cfg.removeFraction < 0. || cfg.removeFraction >= 1.) {
+    ACTS_ERROR("value of removeFraction is "
+               << cfg.removeFraction << ", allowed values are between 0 and 1");
+  }
+  if (std::isnan(cfg.minVtxShift)) {
+    ACTS_ERROR("value of minVtxShift was not initialized");
+  }
```

Comment:
I would argue that the `nan` checks can be removed as this is not the default value and it would most likely cause an FPE downstream anyways. also I think it is the users job to not give unreasonable configs

Reply:
Hi @andiwand, I've deleted those `std::isnan`. I copied it from `SeedFinder` but I see now that it's not widely used.
For checks of reasonable values, I would prefer to keep them. It's an easy and fast way to let a user know that something went wrong.

Cheers, Petr